test/data/test_dataset_summary.py

```diff
@@ -15,31 +25,52 @@ def test_dataset_summary():
     assert summary.name == 'FakeDataset'
     assert summary.num_graphs == 10
 
-    assert summary.num_nodes.mean == num_nodes.mean().item()
-    assert summary.num_nodes.std == num_nodes.std().item()
-    assert summary.num_nodes.min == num_nodes.min().item()
-    assert summary.num_nodes.quantile25 == num_nodes.quantile(0.25).item()
-    assert summary.num_nodes.median == num_nodes.median().item()
-    assert summary.num_nodes.quantile75 == num_nodes.quantile(0.75).item()
-    assert summary.num_nodes.max == num_nodes.max().item()
-
-    assert summary.num_edges.mean == num_edges.mean().item()
-    assert summary.num_edges.std == num_edges.std().item()
-    assert summary.num_edges.min == num_edges.min().item()
-    assert summary.num_edges.quantile25 == num_edges.quantile(0.25).item()
-    assert summary.num_edges.median == num_edges.median().item()
-    assert summary.num_edges.quantile75 == num_edges.quantile(0.75).item()
-    assert summary.num_edges.max == num_edges.max().item()
+    check_summary_item(summary.num_nodes, num_nodes)
+    check_summary_item(summary.num_edges, num_edges)
 
 
 @withPackage('tabulate')
 def test_dataset_summary_hetero():
     dataset1 = FakeHeteroDataset(num_graphs=10)
-    summary1 = Summary.from_dataset(dataset1)
+    summary1 = Summary.from_dataset(dataset1, per_type_breakdown=False)
 
     dataset2 = [data.to_homogeneous() for data in dataset1]
     summary2 = Summary.from_dataset(dataset2)
     summary2.name = 'FakeHeteroDataset'
 
     assert summary1 == summary2
     assert str(summary1) == str(summary2)
+
+
+def test_dataset_summary_hetero_per_type_check():
+    dataset = FakeHeteroDataset(num_graphs=10)
+    exp_num_nodes = torch.Tensor([data.num_nodes for data in dataset])
+    exp_num_edges = torch.Tensor([data.num_edges for data in dataset])
+
+    summary = dataset.get_summary()
+
+    assert summary.name == 'FakeHeteroDataset'
+    assert summary.num_graphs == 10
+
+    check_summary_item(summary.num_nodes, exp_num_nodes)
+    check_summary_item(summary.num_edges, exp_num_edges)
+
```

Comment:
small, maybe not that important:
You could consider adding
`assert dataset.node_types`
and
`assert dataset.edge_types` below.
In the future, when someone changes names of those attributes (summary would stop working correctly), this test won't fail without such asserts. Loops will execute 0 iterations and asserts will probably pass.

I guess it would be nice to make this test fail in such case to make it more obvious that something depends on those attributes and they cannot be removed/renamed like that.